src/RA_AchievementSet.cpp

```diff
@@ -421,38 +421,43 @@ BOOL AchievementSet::LoadFromFile(GameID nGameID)
     const std::string sFilename = GetAchievementSetFilename(nGameID);
 
     SetCurrentDirectory(NativeStr(g_sHomeDir).c_str());
-    FILE* pFile = nullptr;
-    errno_t nErr = fopen_s(&pFile, sFilename.c_str(), "r");
-    if (pFile != nullptr)
+
+    std::ios::sync_with_stdio(false);
+
+    if (std::ifstream ifile{ sFilename }; !ifile.is_open())
+    {
+        //	Cannot open file
+        RA_LOG("Cannot open file %s\n", sFilename.c_str());
+        RA_LOG("Error %s\n", ra::GetLastErrorMsg());
+        return FALSE;
+    }
+    else
     {
         //	Store this: we are now assuming this is the correct checksum if we have a file for it
         g_pCurrentGameData->SetGameID(nGameID);
 
         if (m_nSetType == Local)
         {
-            const char EndLine = '\n';
-            char buffer[4096];
-            unsigned long nCharsRead = 0;
+            auto buffer{ std::make_unique<char[]>(4096) };
 
             //	Get min ver:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
+            ifile.getline(buffer.get(), 4096);
+            auto nCharsRead{ ifile.gcount() };
             //	UNUSED at this point? TBD
 
             //	Get game title:
-            _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-            if (nCharsRead > 0)
-            {
-                buffer[nCharsRead - 1] = '\0';	//	Turn that endline into an end-string
+            ifile.getline(buffer.get(), 4096);
 
-                //	Loading Local from file...?
-                g_pCurrentGameData->SetGameTitle(buffer);
-            }
+            //	Loading Local from file...?
+            if (nCharsRead = ifile.gcount(); nCharsRead > 0)
+                g_pCurrentGameData->SetGameTitle(buffer.get());
 
-            while (!feof(pFile))
+            while (!ifile.eof())
             {
-                ZeroMemory(buffer, 4096);
-                _ReadTil(EndLine, buffer, 4096, &nCharsRead, pFile);
-                if (nCharsRead > 0)
+                buffer.reset();
+                buffer = std::make_unique<char[]>(4096);
```

Comment:
A safer way of doing `ZeroMemory`

Reply:
buffer should only be allocated once outside the loop.

Reply:
The `ZeroMemory` is probably unnecessary. The `nCharsRead` will prevent using stale data.